Tests/CADCacheAdvanceTests/CADCacheAdvanceTests.m

```diff
@@ -0,0 +1,616 @@
+@end
+
+@implementation CADCacheAdvanceTests
+
+// MARK: Behavior Tests
+
+- (void)test_fileURL_returnsUnderlyingFileURL;
+{
+    CADCacheAdvance *const cache = [self createCacheThatOverwitesOldMessages:YES];
+    XCTAssertEqualObjects(cache.fileURL, [self testFileLocation]);
+}
+
+- (void)test_isWritable_returnsTrueForAWritableCache;
+{
+    CADCacheAdvance *const cache = [self createCacheThatOverwitesOldMessages:YES];
+    XCTAssertTrue(cache.isWritable);
+}
+
+- (void)test_isEmpty_returnsTrueForAnEmptyCache;
+{
+    CADCacheAdvance *const cache = [self createCacheThatOverwitesOldMessages:YES];
+    XCTAssertTrue(cache.isEmpty);
+}
+
+- (void)test_isEmpty_returnsFalseForANonEmptyCache;
+{
+    CADCacheAdvance *const cache = [self createCacheThatOverwitesOldMessages:YES];
+    NSError *error = nil;
+    [cache appendMessage:[@"Test" dataUsingEncoding:NSUTF8StringEncoding]
+                error:&error];
+    XCTAssertNil(error);
+    XCTAssertFalse(cache.isEmpty);
+}
+
+- (void)test_messagesAndReturnError_returnsAppendedMessage;
+{
+    CADCacheAdvance *const cache = [self createCacheThatOverwitesOldMessages:YES];
+    NSData *const message = [@"Test" dataUsingEncoding:NSUTF8StringEncoding];
+    NSError *error = nil;
+    [cache appendMessage:message
+                   error:&error];
+    XCTAssertNil(error);
+    XCTAssertEqualObjects([cache messagesAndReturnError:nil], @[message]);
+}
+
+// MARK: Performance Tests
+
+- (void)test_performance_append_fillableCache;
+{
+    CADCacheAdvance *const cache = [self createCacheThatOverwitesOldMessages:NO];
+    // Fill the cache before the test starts.
+    for (NSString *const message in [self lorumIpsumMessages]) {
+        [cache appendMessage:[message dataUsingEncoding:NSUTF8StringEncoding]
+                       error:nil];
+    }
+
+    [self measureBlock:^{
+        for (NSString *const message in [self lorumIpsumMessages]) {
+            [cache appendMessage:[message dataUsingEncoding:NSUTF8StringEncoding]
+                           error:nil];
+        }
+    }];
+}
+
+- (void)test_performance_messages_fillableCache;
+{
+    CADCacheAdvance *const cache = [self createCacheThatOverwitesOldMessages:NO];
+    // Fill the cache before the test starts.
+    for (NSString *const message in [self lorumIpsumMessages]) {
+        [cache appendMessage:[message dataUsingEncoding:NSUTF8StringEncoding]
+                       error:nil];
+    }
+
+    [self measureBlock:^{
+        (void)[cache messagesAndReturnError:nil];
+    }];
+}
+
+// MARK: Private
+
+- (CADCacheAdvance *)createCacheThatOverwitesOldMessages:(BOOL)overwritesOldMessages;
+{
+    [NSFileManager.defaultManager createFileAtPath:[self testFileLocation].path
+                                          contents:nil
+                                        attributes:nil];
+    NSError *error = nil;
+    CADCacheAdvance *const cache = [[CADCacheAdvance alloc]
+                                    initWithFileURL:self.testFileLocation
```

Comment:
Could be nice to double check this indentation. I was expecting one more space but I could very well be misremembering.

Reply:
This is the correct one. But I did get it wrong in two other spots you highlighted 🙂